pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -557,6 +583,13 @@ func (r *ReconcileClusterPool) createCluster(
 	}
 	logger.WithField("cluster", ns.Name).Info("Creating new cluster")
 
+	annotations := clp.Spec.Annotations
+	// Annotate the CD with the hash of the (important) pool spec fields so we can distinguish "stale" CDs
```

Comment:
i think we can abstract that to the pool version itself instead of re stating how we get the pool version..